### PR TITLE
Issue-25: Test imagefield model via Api

### PR DESCRIPTION
**Pull request by AltusBarry**

### CODE REVIEW

File: adminapi/tests/__init__.py [hunk 1]
```diff
@@ -810,3 +827,53 @@ def test_API_no_matching_model_name_from_url_error_put(self):
...
+            reverse("test-generic-list", args=["adminapi", "imagemodel"]),
+            {
+                "title": "Image model 2",
+                "image": open(IMAGE_PATH)
+            },
+            format="multipart"
+        )
+        self.assertEqual(response.status_code, 201)
+        # Test that image was saved to directory and then delete it
+        self.assertTrue(os.path.isfile(glob.glob(RES_DIR+"/image_*.jpg")[0]))
```
Comment (hedleyroos) [inline adminapi/tests/__init__.py]: Spaces around the +.

File: adminapi/tests/__init__.py [hunk 2]
```diff
@@ -810,3 +827,53 @@ def test_API_no_matching_model_name_from_url_error_put(self):
...
+        self.assertEqual(parsed_json["id"], 2)
+
+    def test_image_update_API(self):
+        client = APIClient()
+        response = client.put(
+            reverse("test-generic-detail", args=["adminapi", "imagemodel", 1]),
+            {
+                "image": open(IMAGE_PATH)
+            },
+            format="multipart"
```
Comment (hedleyroos) [inline adminapi/tests/__init__.py:875]: Is the multipart name correct? I just know multipart/form-data.

File: adminapi/tests/__init__.py [hunk 3]
```diff
@@ -810,3 +827,53 @@ def test_API_no_matching_model_name_from_url_error_put(self):
...
+        )
+        import pdb
+        # Test that image was saved to directory and then delete it
+        self.assertTrue(os.path.isfile(glob.glob(RES_DIR+"/image_*.jpg")[0]))
```
Comment (hedleyroos) [inline adminapi/tests/__init__.py]: Spaces

File: adminapi/tests/__init__.py [hunk 4]
```diff
@@ -810,3 +827,53 @@ def test_API_no_matching_model_name_from_url_error_put(self):
...
+    def test_image_update_API(self):
+        client = APIClient()
+        response = client.put(
+            reverse("test-generic-detail", args=["adminapi", "imagemodel", 1]),
+            {
+                "image": open(IMAGE_PATH)
+            },
+            format="multipart"
+        )
+        import pdb
```
Comment (hedleyroos) [inline adminapi/tests/__init__.py]: Remove the pdb

File: adminapi/tests/__init__.py (hunk 2 above)
Comment (AltusBarry) [inline adminapi/tests/__init__.py:875]: format="multipart" is compatible with Django's existing RequestFactory. One would need to set TEST_REQUEST_RENDERER_CLASSES in test_settings.py to support more format types.

To use the traditional content_type argument data would need to be explicitly encoded. For multipart/form-data this includes setting the boundary data and adding it to the header.

**Pull request closed without merging**